### PR TITLE
Delete XRT from the main branch

### PR DESCRIPTION
**Pull request by will-cromar**

### CODE REVIEW

File: test/pjrt/test_runtime.py [hunk 1]
```diff
@@ -55,14 +55,12 @@ def test_xla_device_error(self):
...
       'PJRT_DEVICE': 'TPU',
       'XRT_TPU_CONFIG': 'localservice;0;localhost:51011',
-  }, True), ('xrt_gpu', {
+  }, True), ('gpu_num_devives', {
       'GPU_NUM_DEVICES': '4'
-  }, False), ('pjrt_gpu', {
+  }, True), ('pjrt_gpu', {
       'PJRT_DEVICE': 'GPU',
       'GPU_NUM_DEVICES': '4'
   }, True), ('xla_dist_worker', {
```
Comment (ManfeiBai) [inline test/pjrt/test_runtime.py:66]: do we still want to keep this 'xla_dist_worker' test? not sure what this test for and wanna discuss here

File: torch_xla/csrc/init_python_bindings.cpp [hunk 2]
```diff
@@ -38,7 +38,6 @@
 #include "torch_xla/csrc/ir.h"
 #include "torch_xla/csrc/ir_dump_util.h"
 #include "torch_xla/csrc/ops/device_data.h"
-#include "torch_xla/csrc/runtime/mesh_service.h"
```
Comment (ManfeiBai) [inline torch_xla/csrc/init_python_bindings.cpp:41]: Cool!!! Thanks!

File: torch_xla/csrc/runtime/BUILD [hunk 3]
```diff
@@ -20,21 +12,6 @@ exports_files([
...
-)
-
-tf_proto_library_cc(
-    name = "mesh_service_proto",
-    srcs = ["mesh_service.proto"],
-    has_services = 1,
-    cc_api_version = 2,
-    cc_grpc_version = 1,
-    protodeps = [
-        "@org_tensorflow//tensorflow/core/protobuf/tpu:topology_proto",
```
Comment (ManfeiBai) [inline torch_xla/csrc/runtime/BUILD:35]: Cool!!!

**Review by ManfeiBai — APPROVED** [review]
Thanks, Will! very excited to this change!

File: test/pjrt/test_runtime.py (hunk 1 above)
Comment (will-cromar) [reply test/pjrt/test_runtime.py]: Good catch. I missed this on my first pass.

Comment (will-cromar) [reply test/pjrt/test_runtime.py]: This tested that the PJRT device wouldn't be set if you started XRT distributed training with `xla_dist`. This PR removes `xla_dist`, so we can remove this case.

Comment (ManfeiBai) [reply test/pjrt/test_runtime.py]: Thanks for clarification, got it!

**Comment by will-cromar** [comment]
```
+ python3 /tmp/pytorch/xla/test/test_autocast.py
test_autocast_banned (__main__.TestAutocast) ... FAIL
test_autocast_linalg_fp16 (__main__.TestAutocast) ... FAIL
test_autocast_methods_expect_builtin_promote (__main__.TestAutocast) ... ok
test_autocast_methods_fp16 (__main__.TestAutocast) ... FAIL
test_autocast_methods_fp32 (__main__.TestAutocast) ... FAIL
test_autocast_nn_fp16 (__main__.TestAutocast) ... FAIL
test_autocast_nn_fp32 (__main__.TestAutocast) ... FAIL
test_autocast_torch_expect_builtin_promote (__main__.TestAutocast) ... ok
test_autocast_torch_fp32 (__main__.TestAutocast) ... FAIL
test_autocast_torch_need_autocast_promote (__main__.TestAutocast) ... FAIL
```

I couldn't reproduce this locally with an old PyTorch build, and I see the same error on the `xrt` branch (currently a mirror of master): https://github.com/pytorch/xla/actions/runs/5360445307/jobs/9725678891

I think the CI is just flaking.

**Comment by ManfeiBai** [comment]
> ```
> + python3 /tmp/pytorch/xla/test/test_autocast.py
> test_autocast_banned (__main__.TestAutocast) ... FAIL
> test_autocast_linalg_fp16 (__main__.TestAutocast) ... FAIL
> test_autocast_methods_expect_builtin_promote (__main__.TestAutocast) ... ok
> test_autocast_methods_fp16 (__main__.TestAutocast) ... FAIL
> test_autocast_methods_fp32 (__main__.TestAutocast) ... FAIL
> test_autocast_nn_fp16 (__main__.TestAutocast) ... FAIL
> test_autocast_nn_fp32 (__main__.TestAutocast) ... FAIL
> test_autocast_torch_expect_builtin_promote (__main__.TestAutocast) ... ok
> test_autocast_torch_fp32 (__main__.TestAutocast) ... FAIL
> test_autocast_torch_need_autocast_promote (__main__.TestAutocast) ... FAIL
> ```
> 
> I couldn't reproduce this locally with an old PyTorch build, and I see the same error on the `xrt` branch (currently a mirror of master): https://github.com/pytorch/xla/actions/runs/5360445307/jobs/9725678891
> 
> I think the CI is just flaking.

met the almost same error when just rebase openxla migration branch: https://github.com/pytorch/xla/pull/5242

File: torch_xla/__init__.py [hunk 4]
```diff
@@ -2,33 +2,10 @@
 import os
 import re
 import tempfile
-import subprocess
 
 logging.basicConfig()
 logger = logging.getLogger(__name__)
 
-XRT_RUN_SERVER_PROCESS = 'torch_xla.core._xrt_run_server'
-XRT_SERVER_REGEX = '^python3 -m {} [0-9]+$'.format(XRT_RUN_SERVER_PROCESS)
```
Comment (JackCaoG) [inline torch_xla/__init__.py:11]: oh nice lol, I almost forgot I added this code.

File: torch_xla/csrc/init_python_bindings.cpp [hunk 5]
```diff
@@ -516,24 +515,6 @@ py::object GetRevisions() {
   return py_dict;
 }
 
-std::vector<py::bytes> Rendezvous(int ordinal, const std::string& tag,
```
Comment (JackCaoG) [inline torch_xla/csrc/init_python_bindings.cpp:519]: FYI @amithrm we are removing the mesh server based Rendezvous as part of the XRT clean up

**Comment by ManfeiBai** [comment]
do we want to run TPU CI test for this PR too?

**Comment by miladm** [comment]
My deep respect and appreciation for this PR <3

@will-cromar @ManfeiBai @JackCaoG

**Comment by carmocca** [comment]
Hi folks! When will this be released? This will break the Lightning integration, so I'd like to update it in time.

**Comment by will-cromar** [comment]
Hey @carmocca. Sorry for not providing more detail in the PR description. To be clear, we will still be releasing a wheel for 2.1 that does have XRT that will be built from the `xrt` branch. This PR just removes XRT from the main branch so we can move our main build onto OpenXLA, which does not support XRT at all.

**Comment by carmocca** [comment]
@will-cromar Am I understanding correctly that 2.1 will have 2 different wheels, one with XRT + PjRT and another with just PjRT?
If so, will both be under the `torch_xla` package name?
Will there be a recommended way to distinguish them other than calling `hasattr` on the removed code?

I'm asking these questions because in Lightning we detect if XLA is available by importing `torch_xla`, if we want to continue supporting XRT (maybe we don't), then the wheel that doesn't include XRT would not work with Lightning unless we specifically add code paths depending on which wheel was installed.

I would appreciate it if you could notify me when these wheels are in a release candidate stage. I opened an issue on the lightning side: https://github.com/Lightning-AI/lightning/issues/17982 if you want to keep this discussion off the PR.

Thank you!

cc @Liyang90 @gkroiz

**Comment by will-cromar** [comment]
Thanks. I'll move the conversation to the other thread.

**Comment by giuliano-97** [comment]
Hi, I am trying to follow this guide https://github.com/ronghanghu/vit_10b_fsdp_example (also linked in the torch xla docs) to train a model with FSDP on a v3-128 TPU pod. The [last step prescribes](https://github.com/ronghanghu/vit_10b_fsdp_example?tab=readme-ov-file#running-the-experiments) to run something like:
```
python3 -m torch_xla.distributed.xla_dist ...
```
with some clearly tpu specific flags which are not in `torchrun`. That module was removed in this PR. Sorry for the probably nooby question, but was is "the new way" of achieving that? Should I open an issue instead?

**Comment by JackCaoG** [comment]
@giuliano-97 try https://github.com/pytorch/xla/blob/master/docs/pjrt.md#pods. We also introduced FSDPv2 in https://github.com/pytorch/xla/blob/master/docs/fsdpv2.md, you can find a working llama2 example in https://github.com/pytorch-tpu/transformers/blob/alanwaketan/flash_attention/USER_GUIDE.md

**Comment by giuliano-97** [comment]
> @giuliano-97 try [`master`/docs/pjrt.md#pods](https://github.com/pytorch/xla/blob/master/docs/pjrt.md?rgh-link-date=2024-09-03T17%3A32%3A20Z#pods). We also introduced FSDPv2 in [`master`/docs/fsdpv2.md](https://github.com/pytorch/xla/blob/master/docs/fsdpv2.md?rgh-link-date=2024-09-03T17%3A32%3A20Z), you can find a working llama2 example in [pytorch-tpu/transformers@`alanwaketan`/flash_attention/USER_GUIDE.md](https://github.com/pytorch-tpu/transformers/blob/alanwaketan/flash_attention/USER_GUIDE.md?rgh-link-date=2024-09-03T17%3A32%3A20Z)

thank you :)

**Pull request merged**